lib/puppet/file_serving/fileset.rb

```diff
@@ -122,7 +122,7 @@ def basename
     def children
       return [] unless directory?
 
-      Dir.entries(path).
+      Dir.entries(path, encoding: Encoding::UTF_8).
```

Comment:
Could this fail if UTF_8 encoding is not supported on the target machine? Is that a possible scenario? Sorry if that seems like a stupid question, this is a critical resource so it's worth double checking.

Reply:
A github search shows many places in puppet where UTF-8 is assumed to be available and even forced when not already present -- I can't say with 100% certainty that this is okay, but I would have expected other uses of UTF-8 to be failing if it weren't.

Reply:
Cool, thanks for looking.